plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java

```diff
@@ -261,10 +276,20 @@ private static int getIcebergId(OrcType orcColumn)
         private ByteBuffer min;
         private ByteBuffer max;
 
-        private IcebergMinMax(org.apache.iceberg.types.Type type, Object min, Object max)
+        private IcebergMinMax(org.apache.iceberg.types.Type type, Object min, Object max, MetricsModes.MetricsMode metricsMode)
         {
-            this.min = Conversions.toByteBuffer(type, min);
-            this.max = Conversions.toByteBuffer(type, max);
+            // Out of the two types which could be truncated, string or binary, ORC only supports string bounds.
+            // Therefore, truncation will be applied if needed only on string type.
+            if (metricsMode instanceof MetricsModes.Truncate && type.typeId() == org.apache.iceberg.types.Type.TypeID.STRING) {
+                MetricsModes.Truncate truncateMode = (MetricsModes.Truncate) metricsMode;
+                int truncateLength = truncateMode.length();
+                this.min = UnicodeUtil.truncateStringMin(Literal.of((CharSequence) min), truncateLength).toByteBuffer();
+                this.max = UnicodeUtil.truncateStringMax(Literal.of((CharSequence) max), truncateLength).toByteBuffer();
+            }
+            else {
+                this.min = Conversions.toByteBuffer(type, min);
+                this.max = Conversions.toByteBuffer(type, max);
```

Comment:
if MetricsMode is != Full (eg Counts, None), no min/max should be stored.

again, this should be handled in defensive manner, instead of if-ing on a Full class